### ISSUE TITLE
[Spell] Corruption

### ISSUE THREAD

**Issue by Ashamedlol** [WoWManiaUK/Redemption#2708]
Warlock spell Corruption is off-scaling in all specs, but specifically affliction, tested multiple times and recieved final result:

Demo:
Coeff: 1.2 * 3001 SP  =
3601.2 + 10% (Multiplier) + 1080 (Base Value) = 5041.32
Divided into 6 ticks: 840.22
Current: 
858 * 6 ticks = 5149 

----------------------------------------------------
AFFLIC:
Coeff: 1.86 * 3001 SP = 5581.86
      + 39.1% (Multiplier) + 1458 (Base Value) = 9222.36726
Divided into 6 ticks: 9222.36726/6 = 1537.06121

Current:
1995 * 6 ticks = 11970

----------------------------------------------------
Keep in mind Glyph of Quick Decay (corruption scales with haste) was used on both specs at 1030 haste, and Affliction gets different coeff and multiplier due to specific talents.

Spell ID: http://wow-mania.com/armory/?spell=47813

**Comment by Dranah** [comment]
Ok so, after making a thorough test, the result is as follows
Corruption itself isnt broken, Affliction talents however are buggy.

1) No talents: the damage and scaling is correct.
1314 spell power x 1,2 = 1576
1576 + 1080 / 6 = 442 per tick. Number and test matches.

2) Demo build, 10% multiplier from Demonic Pact: Damage and scaling is correct.
1314 spell power x 1,2 = 1576
(1576+1080) x 1,1 / 6 = 486 damage, Number and test matches.
PS: you did a miscalc with demo damage because you only applied the 10% multiplier to the spell power coeff, while it affects the total damage

3) Empowered Corruption works as intended. 
Adds 12% extra coeff per point (2% per tick)
With 1314 spell power that means +26,5 damage per tick per point. Number and test matches.
442 > 469 > 495 > 521 per tick

4) Malediction works as intended

5) Affliction talents that increase Corruption damage by a flat % are all bugged and in a common manner: 
When the talent is taken, an already ticking corruption is properly affected, however when the spell is recast, it gains an additional damage boost out of nowhere. (Note the spell checks for different stats upon recast as between ticks, which could be the root of bugs )
The boost seems to scale with spell power. My wild guess would be that the talent double-dips on the spell power portion of the spell's damage, with the second dip being applied only on recast.

Talents that are affected:

ID 17811 Improved Corruption
ID 63108 Siphon Life
ID 18271 Shadow Mastery
ID 30060 Contagion

**Comment by Ashamedlol** [comment]
Yeah, the multip calc was slightly off but it was in close numbers.
Demo is 858, so the "Current" version is correct

Fixed numbers on the affliction 3001 SP test:
Coeff: 1.86 * 3001 SP = 5581.86
 + 1458 (Base Value) = 7039.86 
+ 39.1% (Multiplier)  = 9792.44526
9792.44526 / 6 = 1632.07421

Though, Corruption is only getting increased damage from procs when refreshed, but I believe that existed in retail along with Nevermelting Ice Crystal.

**Comment by Dranah** [comment]
Ok this gave me a headache to find how exactly the spell calculates right now, because not only the mods are double dipping, but also the stacking method is wrong, which kept distorting my results.
But now I think I got the picture:

Here's the INTENDED formula to calculate damage for corruption with all talents taken except Malediction:
**(BaseDamage + SP x Coeff ) x modifiers = total damage**
Coeff: 1,2 + 0,3 (Everlasting Affliction) + 0,36 (Empowered Corruption)
Modifiers: 10% from 17814, 5% from 63108, 15% from 18275, 5% from 30064 all stacking multiplicatively makes its math go as follows for a warlock with 3002 spell power:
**(1080 + 3002 x 1,86) x 1,1 x x 1,05 x 1,15 x 1,05 = 9293 (1548 per tick)**

Here's the formula that the spell currently uses:
(BaseDamage + SP x Coeff x modifiers ) x modifiers = total damage
With modifiers stacking ADDITIVELY for whatever reason instead of multiplicative, making the modifiers total 1,35 instead of 1,394.
For a warlock with 3002 SP:
(1080 + 3002 x 1,86 x 1,35) x 1,35 = 11632 (1938 per tick)
My in game test is 1937 per tick.

Note: This is the same issue we had with Healing Stream Totem a couple months ago.
I found this thread from trinity, https://github.com/TrinityCore/TrinityCore/issues/14812 where the original problem was that the base value (1080) was unaffected by modifiers - just like our report on HST.
Now, after this was "fixed", the modifiers likely started double dipping on some parts of the spell, just like for HST.
